lib/modules/versioning/hashicorp/convertor.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { hashicorp2npm, npm2hashicorp } from './convertor';
+
+describe('modules/versioning/hashicorp/convertor', () => {
+  test.each`
+    hashicorp           | npm
+    ${'4.2.0'}          | ${'4.2.0'}
+    ${'4.2.0-alpha'}    | ${'4.2.0-alpha'}
+    ${'~> 4.0'}         | ${'^4.0'}
+    ${'~> 4.1'}         | ${'^4.1'}
+    ${'~> 4.0.0'}       | ${'~4.0.0'}
+    ${'~> 4.0.1'}       | ${'~4.0.1'}
+    ${'~> 4.1.0'}       | ${'~4.1.0'}
+    ${'~> 4.1.1'}       | ${'~4.1.1'}
+    ${'~> 4.0.0-alpha'} | ${'~4.0.0-alpha'}
+    ${'>= 4.0'}         | ${'>=4.0'}
+    ${'<= 4.0'}         | ${'<=4.0'}
+    ${'> 4.0'}          | ${'>4.0'}
+    ${'< 4.0'}          | ${'<4.0'}
+    ${'> 4.0, < 5.0'}   | ${'>4.0 <5.0'}
+    ${'~> 2.3.4'}       | ${'~2.3.4'}
+  `(
+    'hashicorp2npm("$hashicorp") === $npm && npm2hashicorp("$npm") === $hashicorp',
+    ({ hashicorp, npm }) => {
+      expect(hashicorp2npm(hashicorp)).toBe(npm);
+      expect(npm2hashicorp(npm)).toBe(hashicorp);
+    }
+  );
+
+  // These are non-reflective cases for hashicorp2npm
+  test.each`
+    hashicorp        | npm
+    ${'~> 4'}        | ${'>=4'}
+    ${'~> v4'}       | ${'>=4'}
+    ${'>= v4.0'}     | ${'>=4.0'}
+    ${'>=4.0'}       | ${'>=4.0'}
+    ${'<=4.0'}       | ${'<=4.0'}
+    ${'= 4.0'}       | ${'4.0'}
+    ${'> 4.0,< 5.0'} | ${'>4.0 <5.0'}
+  `('hashicorp2npm("$hashicorp") === $npm', ({ hashicorp, npm }) => {
+    expect(hashicorp2npm(hashicorp)).toBe(npm);
+  });
+
+  // These are non-reflective cases for npm2hashicorp
+  test.each`
+    hashicorp     | npm
+    ${'~> 4.0'}   | ${'^4'}
```

Comment:
What would the new value be once 5.0.0 is released?

Reply:
These ranges both limit to 4.x.x releases

Reply:
You can see [here](https://github.com/renovatebot/renovate/pull/19652/files#diff-9f1872925310cf22ca0512e20a66c0ff19c7409c8ab0625c99dd23f1f5785ba0R68), that the `getNewValue` function will return an updated range for `~> 5.0`